src/rest.cpp

```diff
@@ -391,12 +393,94 @@ static bool rest_tx(HTTPRequest* req, const std::string& strURIPart)
     }
 
     default: {
-        return RESTERR(req, HTTP_NOT_FOUND, "output format not found (available: " + AvailableDataFormatsString() + ")");
+        return RESTERR(req, HTTP_NOT_FOUND, "Output format not found (available: " + AvailableDataFormatsString() + ")");
+    }
+    }
+    // not reached
+    return true;
+}
+
+
+static bool rest_tx_sendrawtx(HTTPRequest* req, const std::string& strURIPart)
+{
+    if (!CheckWarmup(req))
+        return false;
+    std::string hashStr;
+    const RetFormat rf = ParseDataFormat(hashStr, strURIPart);
+
+    std::string strBody = req->ReadBody();
+    if (std::string::npos == strBody.find("transaction="))
+    {
+        return RESTERR(req, HTTP_BAD_REQUEST, "No transaction parameter found");
+    }
+    std::string strTx = strBody.substr(std::string("transaction=").length());
+
+    // parse parameter
+    CTransaction tx;
+    if(rf == RF_BINARY)
+    {
+        try {
+            CDataStream bin(SER_NETWORK, PROTOCOL_VERSION);
+            bin << strTx;
+            bin >> tx;
+        }
+        catch (const std::exception&) {
+            return RESTERR(req, HTTP_BAD_REQUEST, strprintf("TX decode failed %s",strTx));
+        }
+    }
+    else
+    {
+        if (!DecodeHexTx(tx, strTx))
+            return RESTERR(req, HTTP_BAD_REQUEST, strprintf("TX decode failed %s",strTx));
+    }
+    uint256 hashTx = tx.GetHash();
+
+    CCoinsViewCache &view = *pcoinsTip;
+    const CCoins* existingCoins = view.AccessCoins(hashTx);
+    bool fHaveMempool = mempool.exists(hashTx);
+    bool fHaveChain = existingCoins && existingCoins->nHeight < 1000000000;
+    if (!fHaveMempool && !fHaveChain) {
+        // push to local node and sync with wallets
+        CValidationState state;
+        bool fMissingInputs;
+        if (!AcceptToMemoryPool(mempool, state, tx, false, &fMissingInputs, false, false)) {
```

Comment:
I think it's job of the API user / wallet to check if the fee is too high or not, instead of trusting the node to check this and care about it.

The Travis error regarding "absurdly-high-fee" comes later, not because of this but because of the RPC-call to cross-check the result I do there. I'm aware of that, but it already fails earlier.